src/libopensc/reader-pcsc.c

```diff
@@ -420,7 +440,7 @@ static int refresh_attributes(sc_reader_t *reader)
 		SC_FUNC_RETURN(reader->ctx, SC_LOG_DEBUG_VERBOSE, SC_SUCCESS);
 	}
 
-	reader->flags &= ~(SC_READER_CARD_CHANGED|SC_READER_CARD_INUSE|SC_READER_CARD_EXCLUSIVE);
+	reader->flags &= ~(SC_READER_CARD_CHANGED | SC_READER_CARD_INUSE | SC_READER_CARD_EXCLUSIVE | SC_READER_REMOVED);
```

Comment:
Is this change really needed? `SC_READER_REMOVED` basically means that its ressources are invalid and should not be accessed anymore. This flag is reset in pcsc_disconnect, where also the card handle is freed. Resetting this flag here without renewing the ressources may lead to problems.

Reply:
Makes sense, fixed.